misk-zookeeper/src/main/kotlin/misk/zookeeper/CuratorFrameworkProvider.kt

```diff
@@ -47,6 +68,24 @@ internal class CuratorFrameworkProvider @Inject internal constructor(
           }
           ZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly, clientConfig)
         }
+        .aclProvider(object : ACLProvider {
+          override fun getDefaultAcl(): List<ACL> {
+            // Default ACL allows clients CRUD operations on znodes this service has created.
+            // Service identification comes from the client cert
+            return Collections.singletonList(ACL(DEFAULT_PERMS, AUTH_IDS))
+          }
+
+          override fun getAclForPath(path: String?): List<ACL> {
+            // Shared directories need to be created in such a way that anyone can create them if
```

Comment:
i don't understand why this is necessary to  allow any client

Reply:
The directory might not exist. For example, Larry could be trying to write to `/leases/larry/` but `/leases/` may not exist. We need to ensure Larry creates `/leases` in a way so that in the future Moe can create `/leases/moe` without any issues